admin/server/projects.go

```diff
@@ -24,95 +23,48 @@ func (s *Server) ListProjectsForOrganization(ctx context.Context, req *adminv1.L
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	opts, err := paginationOptions(req.PageToken, int(req.PageSize))
+	if err != nil {
+		return nil, err
+	}
+
 	// If user has ManageProjects, return all projects
 	claims := auth.GetClaims(ctx)
+	projs := make([]*database.Project, 0)
 	if claims.OrganizationPermissions(ctx, org.ID).ManageProjects {
-		projs, err := s.admin.DB.FindProjectsForOrganization(ctx, org.ID)
-		if err != nil {
-			return nil, status.Error(codes.InvalidArgument, err.Error())
-		}
-
-		dtos := make([]*adminv1.Project, len(projs))
-		for i, p := range projs {
-			dtos[i] = s.projToDTO(p, org.Name)
-		}
-
-		return &adminv1.ListProjectsForOrganizationResponse{
-			Projects: dtos,
-		}, nil
+		projs, err = s.admin.DB.FindProjectsForOrganization(ctx, org.ID, opts)
+	} else if claims.OwnerType() == auth.OwnerTypeUser {
```

Comment:
Missing another `else` conditions (for everyone else, i.e. anon users) – should return result of `FindPublicProjectsInOrganization`